### PR TITLE
Navbar mask only added when nav is visible

### LINKED ISSUE #544: *error when the navbar is not displayed
The latest dev build throws a large error (into the console) when you run it on a screen which is small enough that the navbar is removed.

[Error.zip](https://github.com/ScottLogic/d3fc-showcase/files/90261/Error.zip)

### PR DESCRIPTION
**Pull request by rjmcneill**
Relates to #544

### CODE REVIEW

**Comment by djmiley** [comment]
LGTM

**Comment by jdhodges431** [comment]
Should this be done via the model instead?

**Comment by jleft** [comment]
@rjmcneill - did you find the root cause of this error? Just wondering which code and conditions are causing this.

**Comment by rjmcneill** [comment]
@jleft The problem is that the ```layoutWidth``` is not a number when it attempts to set the width of the ```mask-background.```
It only happens in mobile.

I can look at a different solution if you like?

**Comment by jleft** [comment]
> @jleft The problem is that the layoutWidth is not a number when it attempts to set the width of the mask-background.
It only happens in mobile.

If it's not a number, what is it? `auto` I guess?

> I can look at a different solution if you like?

Just wondering if we can fix the root cause of this, as I could potentially see this happening in other places too.

**Comment by jleft** [comment]
> If it's not a number, what is it? `auto` I guess?

Just to clarify, I mean the actual width style on the element (not `layoutWidth`, which I assume is `NaN`).

**Comment by jleft** [comment]
FWIW, if this is to do with dimension calculations going wrong, then I'd be tempted to create a util which wraps the relevant [jQuery width function](http://api.jquery.com/width/).

**Comment by rjmcneill** [comment]
It looks like it has nothing set for the width

width

File: src/assets/js/chart/nav.js [hunk 1]
```diff
@@ -3,6 +3,7 @@ import fc from 'd3fc';
 import util from '../util/util';
 import event from '../event';
 import zoomBehavior from '../behavior/zoom';
+import width from '../util/width';
```
Comment (jleft) [inline src/assets/js/chart/nav.js]: `util` is already imported, so if `src/assets/js/util/util.js` is updated to include `width`:
```js
// ...
import width from './width';

export default {
    domain: domain,
    layout: layout,
    uid: uid,
    width: width
};
```


 then we can remove this line and refer to `util.width` below

(Throttle and debounce probably should have been added too, but was missed in review!)

File: src/assets/js/util/width.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import $ from 'jquery';
+
+// Returns the width of a passed element
```
Comment (jleft) [inline src/assets/js/util/width.js]: I'd remove the comment - the code is self-documenting

**Comment by jdhodges431** [comment]
:+1:

**Comment by jleft** [comment]
:+1:

**Pull request merged**